Automattic/jetpack
# Blogroll: save site recommendations and sync on load

This PR persists the attributes of the blogroll item.
On save the payload is built based on the attributes of each blogroll and is saved under `site_recommendations` option.
On load the `site_recommendations` option is loaded and for any recommendation that is not in the blogroll block a blogroll item block is created and added.

Fixes #
https://github.com/Automattic/wp-calypso/issues/81355

## Proposed changes:

* Add hook for retrieving and managing the save of the `site_recommendations`

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Add the blogroll block to your site
* Save the changes
* Run `jetpack docker wp option get recommendations` in the console. You should see the values persisted
* Change the settings to something else 
```
jetpack docker wp eval "update_option('recommendations', array (
  0 => 
  array (
    'id' => '321',
    'name' => 'test1',
    'icon' => 'https://myprofoundthoughtsblog.files.wordpress.com/2021/06/image.jpg?w=36',
    'url' => 'https://test1.com',
    'description' => 'test1 desc',
  ),
  1 => 
  array (
    'id' => '123',
    'name' => 'test2',
    'icon' => 'p72OiE-BD-p2.jpg?w=36',
    'url' => 'https://test2.com',
    'description' => 'test2 desc',
  ),
) );"
```
* Reload the admin page
* The new values should be added to you block as blogroll item blocks
